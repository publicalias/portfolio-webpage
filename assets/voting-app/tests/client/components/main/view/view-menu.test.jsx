"use strict";

//local imports

const ViewMenu = require("../../../../../scripts/client/components/main/view/view-menu");

const { newPoll } = require("../../../../../schemas");
const { testWrapper } = require("../../../test-helpers");

//global imports

const { deepCopy } = require("all/utilities");
const { newUser } = require("redux/schemas");
const { initTestEvent, initTestSnapshot, reactTests } = require("redux/tests/react-tests");

//utilities

const { testMount, testShallow } = testWrapper(ViewMenu);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(ViewMenu));

//view menu

describe("view menu", () => {

  const testSnapshot = initTestSnapshot(testShallow);

  const testClick = initTestEvent(testMount, "click");

  const dataList = [{ user: newUser({ id: "id-a" }) }, { poll: newPoll({ users: { created: "id-a" } }) }];

  it("should match snapshot (default)", () => testSnapshot(null, { poll: newPoll() }));

  it("should match snapshot (created)", () => testSnapshot(...dataList));

  it("should match snapshot (created, settings)", () => {

    const fullDataList = deepCopy(dataList, [{ view: { settings: true } }]);

    testSnapshot(...fullDataList);

  });

  it("should call viewToggleSettings on click", () => {

    const fn = ["viewToggleSettings", []];

    return testClick(".qa-toggle-settings", dataList, fn);

  });

});
