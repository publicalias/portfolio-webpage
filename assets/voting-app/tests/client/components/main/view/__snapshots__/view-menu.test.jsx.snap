// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view menu should match snapshot (created) 1`] = `
<div
  className="c-view-menu"
>
  <button
    className="c-view-menu__control-btn qa-toggle-settings"
    onClick={[Function]}
  >
    Settings
  </button>
  <ViewControls
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formCheckTitle": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleDelete": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "metaToggleDelete": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollRemoveVote": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleDelete": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "delete": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "meta": Object {
          "delete": false,
        },
        "polls": Array [],
        "user": Object {
          "auth": Object {
            "id": "",
            "provider": "",
          },
          "data": Object {
            "restricted": false,
          },
          "id": "id-a",
          "name": "",
          "type": "auth",
        },
        "view": Object {
          "add": "",
          "delete": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    local={
      Object {
        "poll": Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "id-a",
            "flagged": Array [],
            "hidden": Array [],
          },
          "votes": 0,
        },
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
</div>
`;

exports[`view menu should match snapshot (created, settings) 1`] = `
<div
  className="c-view-menu"
>
  <button
    className="c-view-menu__control-btn qa-toggle-settings"
    onClick={[Function]}
  >
    Settings
  </button>
  <ViewSettings
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formCheckTitle": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleDelete": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "metaToggleDelete": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollRemoveVote": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleDelete": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "delete": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "meta": Object {
          "delete": false,
        },
        "polls": Array [],
        "user": Object {
          "auth": Object {
            "id": "",
            "provider": "",
          },
          "data": Object {
            "restricted": false,
          },
          "id": "id-a",
          "name": "",
          "type": "auth",
        },
        "view": Object {
          "add": "",
          "delete": false,
          "settings": true,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    local={
      Object {
        "poll": Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "id-a",
            "flagged": Array [],
            "hidden": Array [],
          },
          "votes": 0,
        },
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
</div>
`;

exports[`view menu should match snapshot (default) 1`] = `
<div
  className="c-view-menu"
>
  <ViewControls
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formCheckTitle": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleDelete": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "metaToggleDelete": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollRemoveVote": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleDelete": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "delete": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "meta": Object {
          "delete": false,
        },
        "polls": Array [],
        "user": Object {},
        "view": Object {
          "add": "",
          "delete": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    local={
      Object {
        "poll": Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "",
            "flagged": Array [],
            "hidden": Array [],
          },
          "votes": 0,
        },
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
</div>
`;
